src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -366,18 +359,18 @@ OverlayImpl::makeRedirectResponse (PeerFinder::Slot::ptr const& slot,
 }
 
 std::shared_ptr<Writer>
-OverlayImpl::makeErrorResponse (PeerFinder::Slot::ptr const& slot,
+OverlayImpl::makeErrorResponse (std::shared_ptr<PeerFinder::Slot> const& slot,
     http_request_type const& request,
     address_type remote_address,
     std::string text)
 {
-    boost::beast::http::response<boost::beast::http::string_body> msg;
+    boost::beast::http::response<boost::beast::http::empty_body> msg;
     msg.version(request.version());
     msg.result(boost::beast::http::status::bad_request);
+    msg.reason("Bad Request (" + text + ")");
     msg.insert("Server", BuildInfo::getFullVersionString());
     msg.insert("Remote-Address", remote_address.to_string());
     msg.insert(boost::beast::http::field::connection, "close");
-    msg.body() = text;
     msg.prepare_payload();
     return std::make_shared<SimpleWriter>(msg);
```

Comment:
`SimpleWriter` is also deprecated, tho I don't know what replaces it (and obviously not part of this patch).